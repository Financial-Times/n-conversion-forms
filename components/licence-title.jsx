import React from 'react';
import PropTypes from 'prop-types';

export function LicenceTitle ({
	displayName = '',
	isTrial = false,
	isB2cPartnershipLicence = false,
}){
	if(isB2cPartnershipLicence || isTrial) {
		return (
			<h1 className="ncf__header ncf__center">
				{
					(displayName !== '') && (`${displayName} | `)
				}
				{
					isB2cPartnershipLicence
						? 'Welcome to the Financial Times'
						: 'Start your free trial'
				}
			</h1>
		);
	}
	{
		return renderB2BTitle(displayName);
	}

}

function renderB2BTitle (displayName) {
	const companyName = displayName ? displayName : 'Your company';
	const boldTextClassName = (displayName !== '') ? 'ncf__bold-licence-text' : 'ncf__light-licence-text';
	const template =
		<h1 className="ncf__header ncf__center">
			<span className="ncf__light-licence-text">Great news! </span>
			<span className={boldTextClassName}>{companyName}</span>
			<span className="ncf__light-licence-text"> has an FT Subscription you can join</span>
		</h1>;

	return template;
}

LicenceTitle.propTypes = {
	displayName: PropTypes.string,
	isTrial: PropTypes.bool,
	isB2cPartnershipLicence: PropTypes.bool,
};
