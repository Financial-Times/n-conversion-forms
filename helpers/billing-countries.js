import { billingCountries } from 'n-common-static-data';

export { billingCountries };
