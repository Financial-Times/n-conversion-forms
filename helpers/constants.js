const printRegions = {
	cemeaV1: 'CEMEA_V1',
	cemeaV2: 'CEMEA_V2',
	apac: 'APAC',
};

const addressLine1AutoComplete = 'address-line1';
const addressLine2AutoComplete = 'address-line2';
const addressLine3AutoComplete = 'address-line3';

module.exports = {
	printRegions,
	addressLine1AutoComplete,
	addressLine2AutoComplete,
	addressLine3AutoComplete,
};
