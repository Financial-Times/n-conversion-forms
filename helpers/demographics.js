const { demographics } = require('n-common-static-data');

export { demographics };
