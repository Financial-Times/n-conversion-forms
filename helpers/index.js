module.exports = {
	'ncf-countries': require('./ncf-countries'),
	'ncf-common-data': require('./ncf-common-data'),
	cemeaV1ISO: require('./supportedCountries').cemeaV1ISO,
	cemeaV2ISO: require('./supportedCountries').cemeaV2ISO,
	apacISO: require('./supportedCountries').apacISO,
	identifyFTShippingZone: require('./supportedCountries')
		.identifyFTShippingZone,
	countriesSupported: require('./supportedCountries').countriesSupported,
	countriesSupportedISO: require('./supportedCountries').countriesSupportedISO,
	supportedPostcodeValidators: require('./supportedPostcodeValidators')
		.supportedPostcodeValidators,
	allSupportedPostcodeExamples: require('./supportedPostcodeExamples')
		.allSupportedPostcodeExamples,
	demographics: require('./demographics').demographics,
	billingCountries: require('./billing-countries').billingCountries,
};
