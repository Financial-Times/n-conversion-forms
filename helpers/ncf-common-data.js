const _get = require('lodash.get');
const COMMON_DATA = require('n-common-static-data');

module.exports = function ({ hash, fn }) {
	const context = Object.assign({}, { [hash.export]: _get(COMMON_DATA, hash.import) }, this);
	return fn(context);
};
