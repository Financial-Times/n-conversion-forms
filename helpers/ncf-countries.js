const { countries } = require('n-common-static-data').billingCountries;

const MINIMUM_TO_GROUP = 50;
const MINIMUM_TO_SHOW_GROUPS = 2;

module.exports = function ({ hash = {}, fn }) {
	const filter = Array.isArray(hash.filterList) ? hash.filterList : false;
	const value = hash.value;

	let data = countries;

	// Only show countries in the filter
	if (filter) {
		data = data.filter(item => filter.includes(item.code));
	}

	// Select the country
	if (value) {
		data = data.map(item => {
			item.selected = item.code === value;
			return item;
		});
	}

	// Group countries
	if (data.length >= MINIMUM_TO_GROUP) {
		data = groupCountries(data);
	}

	const context = Object.assign({ countries: data }, this);
	return fn(context);
};

/**
 * Produce a frequently used option group
 * @param {Array} countries
 * @return {Array}
 */
function groupCountries (countries) {
	const frequentlyUsed = ['GBR', 'USA', 'JPN', 'FRA', 'CAN'];

	const frequentlyUsedCountries = countries
		.filter(item => frequentlyUsed.includes(item.code))
		.map(item => Object.assign({}, item))
		.sort((a, b) => frequentlyUsed.indexOf(a.code) - frequentlyUsed.indexOf(b.code));

	// Not worth showing return standard countries list
	if (frequentlyUsedCountries.length < MINIMUM_TO_SHOW_GROUPS) {
		return countries;
	}

	// If selected country is in frequently used only leave that one selected
	// otherwise the user is shown the lowest selected option
	const selected = frequentlyUsedCountries.find(item => item.selected);
	if (selected) {
		countries.forEach(item => item.selected = false);
	}

	return [
		{ label: 'Frequently Used', countries: frequentlyUsedCountries },
		{ label: 'Alphabetical', countries }
	];
}
