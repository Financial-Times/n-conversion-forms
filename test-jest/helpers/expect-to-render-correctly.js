import ReactDOMServer from 'react-dom/server';

export const expectToRenderCorrectly = {
	toRenderCorrectly(reactComponent, props) {
		// Render the React component with props.
		const ReactComponentRendered = ReactDOMServer.renderToStaticMarkup(
			reactComponent(props)
		);

		// eslint-disable-next-line jest/no-standalone-expect
		expect(ReactComponentRendered).toMatchSnapshot();

		// This point is reached when the above assertions were successful.
		// The test should therefore always pass, that means it needs to be
		// `true` when used normally, and `false` when `.not` was used.
		return {
			pass: !this.isNot,
		};
	},
};
