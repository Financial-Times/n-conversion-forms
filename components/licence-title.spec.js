import {expectToRenderCorrectly} from '../test-jest/helpers/expect-to-render-correctly';
import {LicenceTitle} from './licence-title';

expect.extend(expectToRenderCorrectly);

describe('LicenceTitle', () => {
	it('renders with default props', () => {
		const props = {};

		expect(LicenceTitle).toRenderCorrectly(props);
	});

	it('renders with custom display name', () => {
		const props = { displayName: 'Display name text' };

		expect(LicenceTitle).toRenderCorrectly(props);
	});

	it('renders if is trial', () => {
		const props = { isTrial: true };

		expect(LicenceTitle).toRenderCorrectly(props);
	});

	it('renders if is isB2cPartnershipLicence', () => {
		const props = { isB2cPartnershipLicence: true };

		expect(LicenceTitle).toRenderCorrectly(props);
	});
});
