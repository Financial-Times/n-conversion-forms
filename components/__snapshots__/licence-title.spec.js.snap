// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenceTitle renders if is isB2cPartnershipLicence 1`] = `
<h1 class="ncf__header ncf__center">
  Welcome to the Financial Times
</h1>
`;

exports[`LicenceTitle renders if is trial 1`] = `
<h1 class="ncf__header ncf__center">
  Start your free trial
</h1>
`;

exports[`LicenceTitle renders with custom display name 1`] = `
<h1 class="ncf__header ncf__center">
  <span class="ncf__light-licence-text">
    Great news!
  </span>
  <span class="ncf__bold-licence-text">
    Display name text
  </span>
  <span class="ncf__light-licence-text">
    has an FT Subscription you can join
  </span>
</h1>
`;

exports[`LicenceTitle renders with default props 1`] = `
<h1 class="ncf__header ncf__center">
  <span class="ncf__light-licence-text">
    Great news!
  </span>
  <span class="ncf__light-licence-text">
    has an FT Subscription you can join
  </span>
</h1>
`;
